### ISSUE TITLE
Engineserver HTTP Engine

### ISSUE THREAD

**Issue by Sh4d0wHunt3rX** [blacklanternsecurity/bbot#1693]
![image](https://github.com/user-attachments/assets/02f13c8d-5aee-4e95-b200-dfe245f80e09)

**Comment by TheTechromancer** [comment]
```
ERROR    bbot.core.dnsengine:engine.py:592 EngineServer DNSEngine: Unhandled exception in finished_tasks(set()): Set of coroutines/Futures is empty.
TRACE    bbot.core.dnsengine:engine.py:593 Traceback (most recent call last):
  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 583, in finished_tasks
    done, pending = await asyncio.wait(child_tasks, return_when=asyncio.FIRST_COMPLETED, timeout=timeout)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/tasks.py", line 392, in wait
Error:     raise ValueError('Set of coroutines/Futures is empty.')
ValueError: Set of coroutines/Futures is empty.

ERROR    bbot.core.dnsengine:engine.py:456 EngineServer DNSEngine: error in resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]}): Set of coroutines/Futures is empty.
TRACE    bbot.core.dnsengine:engine.py:457 Traceback (most recent call last):
  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 446, in run_and_yield
    async for _ in command_fn(*args, **kwargs):
  File "/home/runner/work/bbot/bbot/bbot/core/helpers/dns/engine.py", line 393, in resolve_raw_batch
    finished = await self.finished_tasks(client_id, timeout=120)
  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 583, in finished_tasks
    done, pending = await asyncio.wait(child_tasks, return_when=asyncio.FIRST_COMPLETED, timeout=timeout)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/tasks.py", line 392, in wait
Error:     raise ValueError('Set of coroutines/Futures is empty.')
ValueError: Set of coroutines/Futures is empty.

TRACE    bbot.core.dnsengine:engine.py:84 EngineServer DNSEngine: reached end of run-and-yield iteration for resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]})
TRACE    bbot.core.dnshelper:engine.py:84 EngineClient DNSHelper: resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]}) got iteration: {'_e': ("EngineServer DNSEngine: error in resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]}): Set of coroutines/Futures is empty.", 'Traceback (most recent call last):\n  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 446, in run_and_yield\n    async for _ in command_fn(*args, **kwargs):\n  File "/home/runner/work/bbot/bbot/bbot/core/helpers/dns/engine.py", line 393, in resolve_raw_batch\n    finished = await self.finished_tasks(client_id, timeout=120)\n  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 583, in finished_tasks\n    done, pending = await asyncio.wait(child_tasks, return_when=asyncio.FIRST_COMPLETED, timeout=timeout)\n  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/tasks.py", line 392, in wait\n    raise ValueError(\'Set of coroutines/Futures is empty.\')\nValueError: Set of coroutines/Futures is empty.\n')}
TRACE    bbot.core.dnshelper:engine.py:84 EngineClient DNSHelper: got error message: {'_e': ("EngineServer DNSEngine: error in resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]}): Set of coroutines/Futures is empty.", 'Traceback (most recent call last):\n  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 446, in run_and_yield\n    async for _ in command_fn(*args, **kwargs):\n  File "/home/runner/work/bbot/bbot/bbot/core/helpers/dns/engine.py", line 393, in resolve_raw_batch\n    finished = await self.finished_tasks(client_id, timeout=120)\n  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 583, in finished_tasks\n    done, pending = await asyncio.wait(child_tasks, return_when=asyncio.FIRST_COMPLETED, timeout=timeout)\n  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/tasks.py", line 392, in wait\n    raise ValueError(\'Set of coroutines/Futures is empty.\')\nValueError: Set of coroutines/Futures is empty.\n')}
TRACE    bbot.core.dnshelper:engine.py:84 EngineClient DNSHelper: raising DNSError
ERROR    bbot.scanner:scanner.py:1195 Error in dnsresolve.handle_event((DNS_NAME("one.one.one.one", module=TARGET, tags={'target', 'in-scope', 'subdomain'}), {})): /home/runner/work/bbot/bbot/bbot/core/engine.py:146:check_error(): EngineServer DNSEngine: error in resolve_raw_batch((), {'queries': [('one.one.one.one', 'A'), ('one.one.one.one', 'AAAA'), ('one.one.one.one', 'SRV'), ('one.one.one.one', 'MX'), ('one.one.one.one', 'NS'), ('one.one.one.one', 'SOA'), ('one.one.one.one', 'CNAME'), ('one.one.one.one', 'TXT')]}): Set of coroutines/Futures is empty.
TRACE    bbot.scanner:scanner.py:1196 Traceback (most recent call last):
  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 446, in run_and_yield
    async for _ in command_fn(*args, **kwargs):
  File "/home/runner/work/bbot/bbot/bbot/core/helpers/dns/engine.py", line 393, in resolve_raw_batch
    finished = await self.finished_tasks(client_id, timeout=120)
  File "/home/runner/work/bbot/bbot/bbot/core/engine.py", line 583, in finished_tasks
    done, pending = await asyncio.wait(child_tasks, return_when=asyncio.FIRST_COMPLETED, timeout=timeout)
  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/tasks.py", line 392, in wait
Error:     raise ValueError('Set of coroutines/Futures is empty.')
ValueError: Set of coroutines/Futures is empty.
```

**Comment by TheTechromancer** [comment]
@amiremami what branch/version were you on when this happened?

**Comment by Sh4d0wHunt3rX** [comment]
BBOT v2.0.0.4569rc

**Comment by TheTechromancer** [comment]
Fixed in [49c3c7b](https://github.com/blacklanternsecurity/bbot/pull/1678/commits/49c3c7b3cf2e560100c8d76a7563cb7893525292).
